ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java

```diff
@@ -62,34 +91,46 @@
    * @param accountStatsStore The {@link AccountStatsStore} to retrieve stats and store aggregated stats.
    * @param callback a callback which will be invoked when the aggregation report has been generated successfully.
    * @param clusterMapConfig the {@link ClusterMapConfig} associated with helix participant.
+   * @param registry the {@link MetricRegistry}.
    */
   MySqlReportAggregatorTask(HelixManager manager, long relevantTimePeriodInMs, StatsReportType statsReportType,
-      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig) {
+      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig,
+      MetricRegistry registry) {
     this.manager = manager;
     clusterAggregator = new HelixClusterAggregator(relevantTimePeriodInMs);
     this.statsReportType = statsReportType;
     this.accountStatsStore = accountStatsStore;
     this.callback = callback;
     this.clusterMapConfig = clusterMapConfig;
+    this.metrics = new Metrics(registry);
```

Comment:
Wouldn't this cause any problem? We have `AccountStats` and `PartitionClassStats` to collect and two aggregation tasks will be created. For each task, we register all 4 metrics (line 74 ~ 81). Would second aggregation task throw exception something like:  `***metric has been registered already` ?

Reply:
Yes, you are right about the two tasks. But the creating a histogram is "getOrAdd". So if it's also being added to the registry, then it will not be added again.

Reply:
Got it, I have no concern now.